### PR TITLE
Call lintr library before linting

### PR DESCRIPTION
**Pull request by michaelquinn32**
This solves namespace issues related to the objects used to set linting options.

### CODE REVIEW

File: ale_linters/r/lintr.vim [hunk 1]
```diff
@@ -18,3 +21,4 @@ call ale#linter#Define('r', {
 \   'callback': 'ale#handlers#gcc#HandleGCCFormat',
 \   'output_stream': 'both',
 \})
+
```
Comment (w0rp) [inline ale_linters/r/lintr.vim]: You left an extra blank line at the end of the file.

File: ale_linters/r/lintr.vim [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+let g:ale_r_lintr_options = get(g:, 'ale_r_lintr_options', 'with_defaults()')
 " A reasonable alternative default:
-" \   get(g:, 'ale_r_lintr_options', 'lintr::with_defaults(object_usage_linter = NULL)')
+"   get(g:, 'ale_r_lintr_options', 'with_defaults(object_usage_linter = NULL)')
 
 function! ale_linters#r#lintr#GetCommand(buffer) abort
+    let l:cmd_string = 'suppressPackageStartupMessages(library(lintr));'
+          \ . 'lint(cache = FALSE, commandArgs(TRUE),'
+          \ . ale#Var(a:buffer, 'r_lintr_options') . ')'
     return ale#path#BufferCdString(a:buffer)
```
Comment (w0rp) [inline ale_linters/r/lintr.vim]: Put a blank line before `return`.

File: ale_linters/r/lintr.vim [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-\   get(g:, 'ale_r_lintr_options', 'lintr::with_defaults()')
+let g:ale_r_lintr_options = get(g:, 'ale_r_lintr_options', 'with_defaults()')
 " A reasonable alternative default:
-" \   get(g:, 'ale_r_lintr_options', 'lintr::with_defaults(object_usage_linter = NULL)')
+"   get(g:, 'ale_r_lintr_options', 'with_defaults(object_usage_linter = NULL)')
 
 function! ale_linters#r#lintr#GetCommand(buffer) abort
+    let l:cmd_string = 'suppressPackageStartupMessages(library(lintr));'
+          \ . 'lint(cache = FALSE, commandArgs(TRUE),'
+          \ . ale#Var(a:buffer, 'r_lintr_options') . ')'
```
Comment (w0rp) [inline ale_linters/r/lintr.vim]: Indent these lines just like the lines below, and in the rest of the codebase.

**Comment by michaelquinn32** [comment]
Thanks for the feedback! Please let me know if there is more that I can do.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**